src/Workspaces/CSharp/Portable/Rename/CSharpRenameRewriterLanguageService.cs

```diff
@@ -121,33 +98,30 @@ private void AddModifiedSpan(TextSpan oldSpan, TextSpan newSpan)
             {
-                _documentId = parameters.Document.Id;
-                _renameRenamableSymbolDeclaration = parameters.RenamedSymbolDeclarationAnnotation;
+                var document = parameters.Document;
+                _documentId = document.Id;
                 _solution = parameters.OriginalSolution;
-                _replacementText = parameters.ReplacementText;
-                _originalText = parameters.OriginalText;
-                _possibleNameConflicts = parameters.PossibleNameConflicts;
-                _renameLocations = parameters.RenameLocations;
                 _conflictLocations = parameters.ConflictLocationSpans;
-                _cancellationToken = parameters.CancellationToken;
                 _semanticModel = parameters.SemanticModel;
-                _renamedSymbol = parameters.RenameSymbol;
-                _replacementTextValid = parameters.ReplacementTextValid;
+                _cancellationToken = parameters.CancellationToken;
                 _renameSpansTracker = parameters.RenameSpansTracker;
-                _isRenamingInStrings = parameters.IsRenamingInStrings;
-                _isRenamingInComments = parameters.IsRenamingInComments;
-                _stringAndCommentTextSpans = parameters.StringAndCommentTextSpans;
-                _renameAnnotations = parameters.RenameAnnotations;
 
-                _aliasSymbol = _renamedSymbol as IAliasSymbol;
-                _renamableDeclarationLocation = _renamedSymbol.Locations.FirstOrDefault(loc => loc.IsInSource && loc.SourceTree == _semanticModel.SyntaxTree);
-                _isVerbatim = _replacementText.StartsWith("@", StringComparison.Ordinal);
+                _simplificationService = document.GetRequiredLanguageService<ISimplificationService>();
+                _semanticFactsService = document.GetRequiredLanguageService<ISemanticFactsService>();
+                _syntaxFactsService = document.GetRequiredLanguageService<ISyntaxFactsService>();
+
+                _annotatedIdentifierTokens = new();
+                _invocationExpressionsNeedingConflictChecks = new();
+                _renameAnnotations = parameters.RenameAnnotations;
+                _modifiedSubSpans = new();
+                _isProcessingComplexifiedSpans = false;
+                _skipRenameForComplexification = 0;
 
-                _simplificationService = parameters.Document.Project.Services.GetRequiredService<ISimplificationService>();
-                _semanticFactsService = parameters.Document.Project.Services.GetRequiredService<ISemanticFactsService>();
+                _renameContexts = GroupRenameContextBySymbolKey(parameters.RenameSymbolContexts, SymbolKey.GetComparer());
+                _textSpanToRenameContexts = GroupTextRenameContextsByTextSpan(parameters.TokenTextSpanRenameContexts);
+                _stringAndCommentRenameContexts = GroupStringAndCommentsTextSpanRenameContexts(parameters.StringAndCommentsTextSpanRenameContexts);
```

Comment:
These don't depend on the document right, but we're still doing this every single time we're rewriting a document?

Reply:
Yes, but can we put this part as the following work item? (I can put comment here)
The initialization part in Conflict.Session needs to be refactoring

Reply:
Sure, absolutely.

Reply:
I have added comments